plugins/azure/keyvaults/allowedCertificateKeyTypes.js

```diff
@@ -0,0 +1,84 @@
+    link: 'https://docs.microsoft.com/en-us/azure/key-vault/certificates/certificate-access-control',
+    apis: ['vaults:list', 'vaults:getCertificates', 'getCertificatePolicy:get'],
+    settings: {
+        allowed_certificate_key_types: {
+            name: 'Key Vault Certificate Key Types',
+            description: 'Key types supported for certificates in Azure Key Vault',
+            regex: '^(RSA|EC)$',
+            default: 'RSA'
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+        var config = {
+            allowed_certificate_key_types: settings.allowed_certificate_key_types || this.settings.allowed_certificate_key_types,
+        };
+
+        async.each(locations.vaults, function(location, rcb) {
+            var vaults = helpers.addSource(cache, source,
+                ['vaults', 'list', location]);
+
+            if (!vaults) return rcb();
+
+            if (vaults.err || !vaults.data) {
+                helpers.addResult(results, 3, 'Unable to query for Key Vaults: ' + helpers.addError(vaults), location);
+                return rcb();
+            }
+
+            if (!vaults.data.length) {
+                helpers.addResult(results, 0, 'No Key Vaults found', location);
+                return rcb();
+            }
+
+            vaults.data.forEach((vault) => {
+                var certificates = helpers.addSource(cache, source,
+                    ['vaults', 'getCertificates', location, vault.id]);
+
+                if (!certificates || certificates.err || !certificates.data) {
+                    helpers.addResult(results, 3, 'Unable to query for Key Vault certificates: ' + helpers.addError(certificates), location, vault.id);
+                } else if (!certificates.data.length) {
+                    helpers.addResult(results, 0, 'No Key Vault Certificates found', location, vault.id);
+                } else {
+                    certificates.data.forEach((certificate) => {
+                        var certificatePolicy = helpers.addSource(cache, source,
+                            ['getCertificatePolicy', 'get', location, certificate.id]);
+
+                        if (!certificatePolicy || certificatePolicy.err || !certificatePolicy.data) {
+                            helpers.addResult(results, 3, 'Unable to query for Certificate Policy: ' + helpers.addError(certificatePolicy), location, certificate.id);
+                        } else {
+                            const certificateKeys = certificatePolicy.data.key_props;
+                            if (certificateKeys && certificateKeys.kty) {
+                                const allowedCertTypesRegex = new RegExp(config.allowed_certificate_key_types.regex);
+
+                                if (allowedCertTypesRegex.test(certificateKeys.kty)) {
```

Comment:
let's take the setting as comma separate values and see if certificateKeys.kty is in that i.e. config.allowed_certificate_key_type.toLowerCase().includes(certificateKeys.kty.toLowerCase())